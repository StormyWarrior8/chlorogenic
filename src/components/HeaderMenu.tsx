import React from "react"
import { Box, Flexbox } from "../UX"
import styled from "../UX/Styled"

const HeaderContainer = styled(Flexbox)`
  padding: 0em 1em;
  align-items: center;
  background: ${({ theme }) => theme.backgroundColor};
  height: 3em;
`

const Left = styled(Box)`
  flex-grow: 1;
`
const Center = styled(Box)`
  flex-grow: 1;
`
const Right = styled(Flexbox)`
  flex-grow: 1;
  margin-left: auto;
  justify-content: flex-end;
`

export interface HeaderMenuProps {
  left?: React.ReactNode
  center?: React.ReactNode
  right?: React.ReactNode
}

export default ({ left, center, right }: HeaderMenuProps) => {
  return (
    <HeaderContainer>
      <Left>{left}</Left>
      <Center>{center}</Center>
      <Right>{right}</Right>
    </HeaderContainer>
  )
}
