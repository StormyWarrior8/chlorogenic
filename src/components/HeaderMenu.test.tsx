import * as React from "react"
import * as renderer from "react-test-renderer"
import HeaderMenu from "./HeaderMenu"

it("snapshot", () => {
  {
    const tree = renderer.create(<HeaderMenu />).toJSON()
    expect(tree).toMatchSnapshot()
  }
  {
    const tree = renderer
      .create(
        <HeaderMenu
          left={<p>left</p>}
          right={<p>right</p>}
          center={<p>center</p>}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  }
})
