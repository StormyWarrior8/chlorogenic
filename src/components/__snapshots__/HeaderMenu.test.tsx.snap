// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="sc-bxivhb ewuPkI sc-bwzfXH jIEWCi"
>
  <div
    className="sc-ifAKCX ezQlwD sc-bdVaJa fpcbkF"
  />
  <div
    className="sc-EHOje cTZJmS sc-bdVaJa fpcbkF"
  />
  <div
    className="sc-bZQynM gRKxIf sc-bwzfXH jIEWCi"
  />
</div>
`;

exports[`snapshot 2`] = `
<div
  className="sc-bxivhb ewuPkI sc-bwzfXH jIEWCi"
>
  <div
    className="sc-ifAKCX ezQlwD sc-bdVaJa fpcbkF"
  >
    <p>
      left
    </p>
  </div>
  <div
    className="sc-EHOje cTZJmS sc-bdVaJa fpcbkF"
  >
    <p>
      center
    </p>
  </div>
  <div
    className="sc-bZQynM gRKxIf sc-bwzfXH jIEWCi"
  >
    <p>
      right
    </p>
  </div>
</div>
`;
